network/connect_wifi.py

```diff
@@ -0,0 +1,52 @@
+import os
+import pexpect
+import sys
+
+size=len(sys.argv)
+passphrase=sys.argv[size-1]
+passphrase.strip(' ')
+ssid=sys.argv[1]
+if size>=4:
+    i=size-2
+    for loop in range(1,i):
+        ssid=ssid+' '+sys.argv[loop+1]
+        print ssid
+
+cmd_arg=', '.join(sys.argv[1:])
+print cmd_arg
+connmanctl=pexpect.spawn('connmanctl')
+connmanctl.expect('connmanctl>')
+connmanctl.sendline('agent on')
+err=connmanctl.expect(['Agent registered', 'Error'])
+if err == 1:
+    print 'Error in agent on'
+f=open('/usr/share/cockpit/network/output.txt','r')
```

Comment:
Btw, cockpit has a d-bus bridge that can be used directly from JS. See the network implementation fro NetworkManager.... the entire client side can be done on the JS end.